base/range.jl

```diff
@@ -924,9 +924,15 @@ function getindex(r::LinRange{T}, s::OrdinalRange{S}) where {T, S<:Integer}
     end
 end
 
-show(io::IO, r::AbstractRange) = print(io, repr(first(r)), ':', repr(step(r)), ':', repr(last(r)))
 show(io::IO, r::UnitRange) = print(io, repr(first(r)), ':', repr(last(r)))
 show(io::IO, r::OneTo) = print(io, "Base.OneTo(", r.stop, ")")
+function show(io::IO, r::AbstractRange)
+    if step(r) != 0
+        print(io, repr(first(r)), ':', repr(step(r)), ':', repr(last(r)))
+    else
+        print(io, "StepRangeLen(", repr(first(r)), ", ", repr(step(r)), ", ", repr(length(r)), ")")
+    end
+end
```

Comment:
It's ugly to print what seems like an internal detail, but it feels wrong to leave it printing `0:0:0` if we support the use of such ranges, more than by accident. Perhaps it can become something prettier like `range(1, step=0, length=3)` in another PR?

Reply:
> Perhaps it can become something prettier like range(1, step=0, length=3) in another PR?

Yeah, that was my thinking. Since it is not directly related to this PR, I would prefer not to change this here.

Reply:
But what harm in printing something for now? It's only the zero step case. Maybe someone's doctests want that `range(1.0, step=0, length=4)` prints `1.0:0.0:1.0` and you'd rather not break them twice?

Reply:
It seems confusing to have this as a fallback for all `AbstractRange`s, since people may define their own ranges with zero step. Perhaps it would be ok to just print `StepRangeLen` this way?

Reply:
OK, that will work, I've restricted it to StepRangeLen.